services/provider/server/server.go

```diff
@@ -258,10 +259,85 @@ func (s *OCSProviderServer) GetStorageConfig(ctx context.Context, req *pb.Storag
 
-	//TODO: Implement this in a different PR
-	return &pb.GetDesiredClientStateResponse{}, nil
+	// Get storage consumer resource using UUID
+	consumer, err := s.consumerManager.Get(ctx, req.StorageConsumerUUID)
+	if err != nil {
+		return nil, err
+	}
+
+	klog.Infof("Found storageConsumer for GetDesiredClientState")
+
+	// Verify Status
+	switch consumer.Status.State {
+	case ocsv1alpha1.StorageConsumerStateNotEnabled:
+		return nil, status.Errorf(codes.FailedPrecondition, "storageConsumer is in not enabled")
+	case ocsv1alpha1.StorageConsumerStateFailed:
+		return nil, status.Errorf(codes.Internal, "storageConsumer status failed")
+	case ocsv1alpha1.StorageConsumerStateConfiguring:
+		return nil, status.Errorf(codes.Unavailable, "waiting for the resources to be provisioned")
+	case ocsv1alpha1.StorageConsumerStateDeleting:
+		return nil, status.Errorf(codes.NotFound, "storageConsumer is in deleting phase")
+	case ocsv1alpha1.StorageConsumerStateReady:
+		kubeResources, err := s.getKubeResources(ctx, consumer)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "failed to get kube resources. %v", err)
+		}
+
+		response := &pb.GetDesiredClientStateResponse{}
+
+		for i := range kubeResources {
+			kubeResource := kubeResources[i]
+			if kubeResource.GetObjectKind().GroupVersionKind().Group == "" ||
+				kubeResource.GetObjectKind().GroupVersionKind().Kind == "" ||
+				kubeResource.GetName() == "" {
+				klog.Errorf("Resource is not properly structured")
+				return nil, status.Errorf(codes.Internal, "failed to get kube resources.")
+			}
+			sanitizeKubeResource(kubeResource)
+			response.KubeResources = append(response.KubeResources, mustMarshal(kubeResource))
+		}
+
+		channelName, err := s.getOCSSubscriptionChannel(ctx)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "failed to get channel name for Client Operator: %v", err)
+		}
+
+		storageCluster, err := util.GetStorageClusterInNamespace(ctx, s.client, s.namespace)
+		if err != nil {
+			return nil, err
+		}
+
+		inMaintenanceMode, err := s.isSystemInMaintenanceMode(ctx)
+		if err != nil {
+			klog.Error(err)
+			return nil, status.Errorf(codes.Internal, "Failed to get maintenance mode status.")
+		}
+
+		isConsumerMirrorEnabled, err := s.isConsumerMirrorEnabled(ctx, consumer)
+		if err != nil {
+			klog.Error(err)
+			return nil, status.Errorf(codes.Internal, "Failed to get mirroring status for consumer.")
+		}
+
+		desiredClientConfigHash := getDesiredClientConfigHash(
+			channelName,
+			consumer,
+			isEncryptionInTransitEnabled(storageCluster.Spec.Network),
+			inMaintenanceMode,
+			isConsumerMirrorEnabled,
+		)
+		response.DesiredStateHash = desiredClientConfigHash
+		response.ClientOperatorChannel = channelName
+		response.MaintenanceMode = inMaintenanceMode
+		response.MirrorEnabled = isConsumerMirrorEnabled
```

Comment:
Can you please fill in the response field  directly (instead of using a local variable) and in case you cannot, lets at least move the assignment closer to where we are acquiring the information